policy/src/test/java/org/apache/brooklyn/policy/autoscaling/AutoScalerPolicyPoolSizeTest.java

```diff
@@ -0,0 +1,110 @@
+import org.apache.brooklyn.core.entity.EntityAsserts;
+import org.apache.brooklyn.core.entity.trait.Resizable;
+import org.apache.brooklyn.core.test.BrooklynAppUnitTestSupport;
+import org.apache.brooklyn.core.test.entity.TestCluster;
+import org.apache.brooklyn.core.test.entity.TestSizeRecordingCluster;
+import org.apache.brooklyn.entity.stock.BasicStartable;
+import org.apache.brooklyn.util.collections.MutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+public class AutoScalerPolicyPoolSizeTest extends BrooklynAppUnitTestSupport {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AutoScalerPolicyPoolSizeTest.class);
+
+    private static final int CLUSTER_INIITIAL_SIZE = 3;
+    private static final int CLUSTER_MIN_SIZE = 2;
+    private static final int CLUSTER_MAX_SIZE = 4;
+
+    AutoScalerPolicy policy;
+    TestSizeRecordingCluster cluster;
+    List<Integer> resizes = MutableList.of();
+
+    @BeforeMethod(alwaysRun = true)
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        LOG.info("resetting " + getClass().getSimpleName());
+        cluster = app.createAndManageChild(EntitySpec.create(TestSizeRecordingCluster.class)
+                .configure(TestCluster.INITIAL_SIZE, CLUSTER_INIITIAL_SIZE)
+                .configure(TestCluster.MEMBER_SPEC, EntitySpec.create(BasicStartable.class))
+        );
+        PolicySpec<AutoScalerPolicy> policySpec = PolicySpec.create(AutoScalerPolicy.class)
+                .configure(AutoScalerPolicy.RESIZE_OPERATOR, new ResizeOperator() {
+                    @Override
+                    public Integer resize(Entity entity, Integer desiredSize) {
+                        LOG.info("resizing to " + desiredSize);
+                        resizes.add(desiredSize);
+                        return ((Resizable) entity).resize(desiredSize);
+                    }
+                })
+                .configure(AutoScalerPolicy.MIN_POOL_SIZE, CLUSTER_MIN_SIZE)
+                .configure(AutoScalerPolicy.MAX_POOL_SIZE, CLUSTER_MAX_SIZE);
+        policy = cluster.policies().add(policySpec);
+        app.start(ImmutableList.of());
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown() throws Exception {
+        try {
+            if (policy != null) policy.destroy();
+        } finally {
+            super.tearDown();
+            cluster = null;
+            policy = null;
+        }
+    }
+
+    @Test
+    public void testResizeUp() throws Exception {
+        EntityAsserts.assertAttributeEqualsEventually(cluster, TestCluster.GROUP_SIZE, CLUSTER_INIITIAL_SIZE);
+        // Simulate user expunging the entities manually
+        for (int i = 0; i < CLUSTER_MAX_SIZE - CLUSTER_MIN_SIZE; i++) {
```

Comment:
This sum seems weird: for your test it deletes `4 - 2` entities, which takes us from initial size of 3 down to 1. But that feels more like coincidence than design. Do you not want to delete `CLUSTER_INITIAL_SIZE - CLUSTER_MIN_SIZE + 1`?